server/src/logic/middleware/auth-validaton.middleware.ts

```diff
@@ -0,0 +1,129 @@
+import { NextFunction, Request, Response } from "express";
+import * as jwt from "jsonwebtoken";
+import { Config } from "./../../config";
+import { ApiError, ERRORS, ExtractRequestBodyResult, SocoboUser } from "./../../models/index";
+import { UserService } from "./../services/index";
+
+export class AuthValidationMiddleware {
+
+  constructor (private _userService: UserService) {}
+
+  public checkRequest (req: Request): Promise<any> {
```

Comment:
I still think we should do this with the modelValidation. Maybe we need something like a credentials model and add the constraints to the attributes of this model (username, email, password or so). I think so because you still accept all kinds of emails (also wrong ones with out @ and so) and the password and username would also need some additional validation like min length or allowed characters. And I don't think that you want to implement this by yourself ;)

What do you think. Maybe we should adapt this later.